akka-stream/src/main/scala/akka/stream/scaladsl/RetryFlow.scala

```diff
@@ -19,12 +19,16 @@ object RetryFlow {
    * The retry condition is controlled by the `decideRetry` function. It takes the originally emitted
    * element and the response emitted by `flow`, and may return a request to be retried.
    *
-   * The implementation of the `RetryFlow` requires that `flow` follows one-in-one-out semantics,
-   * the [[akka.stream.scaladsl.Flow Flow]] may not filter elements,
-   * nor emit more than one element per incoming element. The `RetryFlow` will fail if two elements are
-   * emitted from the `flow`, it will be stuck "forever" if nothing is emitted. Just one element will
-   * be emitted into the `flow` at any time. The `flow` needs to emit an element before the next
-   * will be emitted to it.
+   * The implementation of the `RetryFlow` requires that `flow` follows a strict first-in-first-out and
+   * one-in-one-out semantics, i.e., the [[akka.stream.scaladsl.Flow Flow]] may neither filter elements,
+   * nor emit more than one element per incoming element. That being said, any sort of batching, grouping, or filtering
```

Comment:
The previous text says that if two elements are emitted, the flow will fail and hang forever if nothing is emitted. The new text is saying that it will hang forever in both cases. Is that correct?